core/flyout_base.js

```diff
@@ -321,6 +322,8 @@ Blockly.Flyout.prototype.init = function(targetWorkspace) {
  */
 Blockly.Flyout.prototype.dispose = function() {
   this.hide();
+  this.workspace_.getComponentManager().removeComponent(
+      'flyout' + this.workspace_.id);
```

Comment:
Since you are creating this id twice, I'd rather it be a property. If anyone ever changes the id when you add the capability there's a decent chance they might forget and break it here (or if someone overrides the corresponding `add` method to change the id, they'd also have to override this just to change the name)

Reply:
Done.